tests/unit/dune_test.py

```diff
@@ -26,3 +37,45 @@ def test_parse_decimal_type(self):
         for invalid in invalid_decimals:
             with self.subTest(msg=invalid):
                 self.assertEqual(tuple([None, None]), _parse_decimal_type(invalid))
+
+    def test__handle_column_types(self):
+        self.assertEqual(
+            tuple([DOUBLE_PRECISION, [], []]), _handle_column_types("real_col", "real")
+        )
+        # decimal(x,y) is handled in a separate test
+        self.assertEqual(
+            tuple([BIGINT, [], []]), _handle_column_types("bigint_col", "bigint")
+        )
+        self.assertEqual(
+            tuple([INTEGER, [], []]), _handle_column_types("int_col", "integer")
+        )
+        self.assertEqual(
+            tuple([BYTEA, ["byte_col"], []]),
+            _handle_column_types("byte_col", "varbinary"),
+        )
+        self.assertEqual(
+            tuple([JSONB, [], ["arr_col"]]),
+            _handle_column_types("arr_col", "unknown_type"),
+        )
+        with patch(
+            "src.sources.dune._parse_decimal_type"
+        ) as _mock_decimal_type, self.assertLogs(level=ERROR) as logs:
+            _mock_decimal_type.return_value = tuple([None, None])
+            _handle_column_types("dec_col", "decimal(12, 2222)")
+            self.assertEqual(
+                tuple([JSONB, [], ["dec_col"]]),
+                _handle_column_types("dec_col", "decimal(12, 2222)"),
+            )
```

Comment:
Nit: Seems like an undesired extra call. So this is JSONB because decimal(X, Y) has invalid X, y values?
```suggestion
            self.assertEqual(
                tuple([JSONB, [], ["dec_col"]]),
                _handle_column_types("dec_col", "decimal(12, 2222)"),
            )
```

Reply:
> So this is JSONB because decimal(X, Y) has invalid X, y values

Yes, we force the decimal parser to return `None, None` for `decimal(X, Y)` where `X` and `Y` would normally be some parsable integers. 
It's simply a test for this failsafe (the `else` clause):
```
        precision, scale = _parse_decimal_type(d_type)
        if precision and scale:
            DUNE_TO_PG[d_type] = NUMERIC(precision, scale)
        else:
            log.error("Failed to parse precision and scale from Dune result: %s", name)
```

Reply:
Don't forget the suggestion here. Remove extra call before assertion.